engines/engine.h

```diff
@@ -611,6 +611,12 @@ class Engine {
 	 */
 	inline Common::SaveFileManager *getSaveFileManager() { return _saveFileMan; }
 
+	/**
+	 * Check if the engine supports a GUI override
+	 * (e.g. in order to display its own quit confirmation dialog)
+	 */
+	virtual bool hasGUIOverride() { return false; }
```

Comment:
This should probably be a feature flag to avoid adding a method for every little exception.
Then you could just do eg. `g_engine->hasFeature(Engine::kSupportsCustomGUI)`

Reply:
Thanks, that's even cleaner! Let me change that...